tests/management/group/test_view.py

```diff
@@ -311,6 +326,46 @@ def test_add_group_roles_system_policy_create_success(self):
         self.assertEqual(roles[0].get('description'), self.role.description)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
+    def test_system_flag_update_on_add(self):
+        """Test that adding a role to a platform_default group flips the system flag."""
+        url = reverse('group-roles', kwargs={'uuid': self.defGroup.uuid})
+        client = APIClient()
+        test_data = {'roles': [self.roleB.uuid, self.dummy_role_id]}
+
```

Comment:
Might be worth adding an assertion before the request as well, just as a sanity check to ensure the initial value?

```
self.assertTrue(self.defGroup.system)
```